internal/provider/engine.go

```diff
@@ -73,3 +79,147 @@ type IncidentEngineExpressionParseOpts struct {
 	Returns IncidentEngineReturnsMeta `tfsdk:"returns"`
 	Source  types.String              `tfsdk:"source"`
 }
+
+// Attributes
```

Comment:
These were defined inline, I've pulled them out so it's easier to read and we can re-use in other resources going forwards